### ISSUE TITLE
Client Components broken in npm workspaces repository

### ISSUE THREAD

**Issue by 6TELOIV** [payloadcms/payload#10009]
### Describe the Bug

This one is a bit of a doozy, so bear with me.

In a npm workspace driven monorepo with different versions of react-dom across the workspaces, an unrecoverable error will occur in client sided custom payload components (including the built-in rich-text-lexical component).

To show that this is not a contrived example: the use case for this is a react native and payload monorepo, were the payload app and react native app are different build targets for the project. React native is still on react 18 and still will be for a long time. I've recreated that setup in this reproduction.

From my debugging, this seems to have to do with `react-dom` in particular, and `useContext` or `createContext` not being defined.

### Link to the code that reproduces this issue

https://github.com/6TELOIV/typeerror-reproduction

### Reproduction Steps

1. Clone the attacher reproduction
2. Run `npm i` from the root of the project
3. Run `npm run dev:server` or `cd apps/next && npm run dev`
4. Try to create an `example` record. Note the error you recieve of `TypeError: _chunk_3MXUY6XL_js__WEBPACK_IMPORTED_MODULE_10__.b() is undefined`
5. Try to edit a `user` or create a `media` record. Note that there is no error.

### Which area(s) are affected? (Select all that apply)

area: core, area: ui

### Environment Info

```text
Payload: 3.7.0
NodeJS: 22.11.0
NextJS: 15.1.0
```

**Comment by 6TELOIV** [comment]
From what I can tell, this has to do in particular with how npm installs the dependencies vs how pnpm does.

npm will place `@payloadcms/ui` and `@payloadcms/rich-text-lexical` in `apps/next/node_modules` because it places the higher react/react-dom deps in the workspace rather than the root.

pnpm will place those in the root `node_modules`, because it puts the older react into `apps/expo/node_modules`.

I'm not 100% certain, but I think this behavior breaks how payload loads and runs the components on the fly.

**Comment by 6TELOIV** [comment]
Yeah, the above logic tracks. I downgraded to 3.2.2, which was working, and npm installs all of the @payloadcms packages as well as payload into the `apps/next/node_modules` directory, and the bug is no longer present. I'll test the other versions in between to figure out exactly when the breaking change occured.

**Comment by 6TELOIV** [comment]
I added another commit in the reproduction repo of how I was able to fix this issue using yarn.

Effectively, using `nodeLinker: node-modules` and `nmHoistingLimits: workspaces`, yarn will let you setup workspaces where their dependencies are installed entirely within the respective workspace, meaning nothing gets hoisted to the root project.

This is a fairly common issue with react native projects, but it may be indicative of payload having some ghost dependencies, maybe between `@payloadcms/next` and `@payloadcms/ui`, or `payload` and `@payloadcms/ui`.

**Comment by JessRynkar** [comment]
Hi! Apologies for the long delay in getting a response here, are you still experiencing this on the latest version `v3.77.0`?

**Comment by 6TELOIV** [comment]
Hi Jess, I am no longer having this issue, likely because the underlying cause is no longer relevant (React Native has caught up, so `react-dom` version mismatches aren't an issue at the moment)

**Comment by JessRynkar** [comment]
Okay that's good to hear, thank you for your report none the less! Closing.
